src/System.Private.CoreLib/src/System/StartupHookProvider.cs

```diff
@@ -85,10 +106,10 @@ private static void CallStartupHook(string assemblyPath)
                 }
                 catch (AmbiguousMatchException)
                 {
-                    // Found multiple
+                    // Found multiple. Will throw below due to initializeMethod being null.
                     Debug.Assert(initializeMethod == null);
-                    wrongSignature = true;
```

Comment:
This actually looks to me like either an incorrect assert that was never hit for some reason, or some kind of bug. The intention is for this case (multiple startup hook methods with incorrect signatures) to throw an ArgumentException, tested in https://github.com/dotnet/core-setup/blob/master/src/test/HostActivationTests/StartupHooks.cs#L435 - I don't know how this test is passing currently.

Reply:
The assert is a bit redundant, but not wrong - it should be null when we get there.
This line removal was actually suggested by the compiler - it marked the line as unnecessary. Which is true, the `wrongSignature` is not looked at if `initializeMethod` is `null`.